### PR TITLE
Camera support

### PR DESCRIPTION
**Pull request by hamxa152**

### CODE REVIEW

File: lib/widget/camera_widget.dart [hunk 1]
```diff
@@ -0,0 +1,438 @@
...
+  bool isFrontCamera = false;
+
+  List<CameraDescription>? cameras;
+  CameraController? cameracontroller;
+
+  final ImagePicker imagePicker = ImagePicker();
+  List<XFile> imageFileList = [];
+  List<Uint8List> imagesFileWeb = [];
+
+  Future<void> cameraFunction() async {
```
Comment (vusters) [inline lib/widget/camera_widget.dart:179]: The name doesn't tell me anything about it. This should just be called initCamera() and right under the WidgetState's initState() and combine with the setCamera call.

File: lib/widget/camera_widget.dart [hunk 2]
```diff
@@ -0,0 +1,438 @@
...
+  void setCamera(int i) {
+    cameracontroller = CameraController(cameras![i], ResolutionPreset.max);
+    cameracontroller!.initialize().then((_) {
+      notifyListeners();
+    });
+  }
+
+  void selectImage() async {
+    final List<XFile>? selectImage = await imagePicker.pickMultiImage();
+    if (selectImage!.isNotEmpty) {
```
Comment (vusters) [inline lib/widget/camera_widget.dart:199]: don't use !. The selectImage also cannot be null according to the function call.

**Comment by vusters** [comment]
1. There are zero comments on what the code does except header for each function. Can you add them?
2. Long blocks of code. Break them up into small functions.
3. Naming is inconsistent. We can work through this later.

File: lib/widget/camera_widget.dart [hunk 3]
```diff
@@ -0,0 +1,438 @@
...
+  List<CameraDescription>? cameras;
+  CameraController? cameracontroller;
+
+  final ImagePicker imagePicker = ImagePicker();
+  List<XFile> imageFileList = [];
+  List<Uint8List> imagesFileWeb = [];
+
+  Future<void> cameraFunction() async {
+    cameras = await availableCameras();
+    notifyListeners();
```
Comment (vusters) [inline lib/widget/camera_widget.dart:181]: What happened when there is no camera or the users reject permission. There is nothing displayed to the users.

File: lib/widget/camera_widget.dart [hunk 4]
```diff
@@ -0,0 +1,438 @@
...
+  }
+
+  void initCamera() {
```
[the remark this answers is not in the document]
Comment (hamxa152) [reply lib/widget/camera_widget.dart:184]: done

File: lib/widget/camera_widget.dart (hunk 1 above)
Comment (hamxa152) [reply lib/widget/camera_widget.dart:179]: fixed

File: lib/widget/camera_widget.dart (hunk 2 above)
Comment (hamxa152) [reply lib/widget/camera_widget.dart:193]: fixed

File: lib/widget/camera_widget.dart [hunk 5]
```diff
@@ -39,30 +40,36 @@ class Camera extends StatefulWidget
     return {
       'height': (value) => _controller.height = Utils.optionalDouble(value),
       'width': (value) => _controller.width = Utils.optionalDouble(value),
+      'ImagePreviewheight': (value) =>
+          _controller.ImagePreviewheight = Utils.optionalDouble(value),
+      'ImagePreviewwidth': (value) =>
+          _controller.ImagePreviewwidth = Utils.optionalDouble(value),
+      'fullImageWidth': (value) =>
```
Comment (vusters) [inline lib/widget/camera_widget.dart:47]: Why is this needed? Just size to the camera's width/height.

File: lib/widget/camera_widget.dart [hunk 6]
```diff
@@ -39,30 +40,36 @@ class Camera extends StatefulWidget
...
+          _controller.fullImageWidth = Utils.optionalDouble(value),
+      'fullImageheight': (value) =>
+          _controller.fullImageheight = Utils.optionalDouble(value),
       'cameraRotateIcon': (value) =>
```
Comment (vusters) [inline lib/widget/camera_widget.dart:51]: dont' expose this

File: lib/widget/camera_widget.dart [hunk 7]
```diff
@@ -39,30 +40,36 @@ class Camera extends StatefulWidget
...
           _controller.cameraRotateIcon = Utils.getIcon(value),
       'imageTakeIcon': (value) =>
```
Comment (vusters) [inline lib/widget/camera_widget.dart:53]: don't expose this

**Review by vusters — COMMENTED** [review]
Don't expose any button stylings. It's overwhelming to tell which is which. We can do it some better way later. Either use our Camera with default actions, or build your own actions.

File: lib/widget/camera_widget.dart (hunk 3 above)
Comment (hamxa152) [reply lib/widget/camera_widget.dart]: when there is no camera i have made the UI for that

File: lib/widget/camera_widget.dart [hunk 8]
```diff
@@ -39,30 +40,36 @@ class Camera extends StatefulWidget
     return {
       'height': (value) => _controller.height = Utils.optionalDouble(value),
       'width': (value) => _controller.width = Utils.optionalDouble(value),
+      'ImagePreviewheight': (value) =>
```
[the remark this answers is not in the document]
Comment (hamxa152) [reply lib/widget/camera_widget.dart]: this is basically full image view when user select image form gallery or capture image form camera. if user can change preview image height

File: lib/widget/camera_widget.dart (hunk 5 above)
Comment (hamxa152) [reply lib/widget/camera_widget.dart:47]: this is basically full image view when user select image form gallery or capture image form camera. if user can change preview image height

File: lib/widget/camera_widget.dart (hunk 7 above)
Comment (hamxa152) [reply lib/widget/camera_widget.dart:53]: naming is consistence please check it

File: .android/app/src/main/AndroidManifest.xml [hunk 9]
```diff
@@ -7,7 +7,6 @@
          to allow setting breakpoints, to provide hot reload, etc.
     -->
     <uses-permission android:name="android.permission.INTERNET"/>
-    <uses-permission android:name="android.permission.ACCESS_FINE_LOCATION" />
```
[the remark this answers is not in the document]
Comment (hamxa152) [reply .android/app/src/main/AndroidManifest.xml]: i added location permission please review

File: .ios/Podfile [hunk 10]
```diff
@@ -31,7 +31,6 @@ target 'Runner' do
   flutter_install_all_ios_pods File.dirname(File.realpath(__FILE__))
 
   pod 'FlutterPluginRegistrant', :path => File.join('Flutter', 'FlutterPluginRegistrant'), :inhibit_warnings => true
-  pod 'FirebaseFirestore', :git => 'https://github.com/invertase/firestore-ios-sdk-frameworks.git', :tag => '10.3.0'
```
[the remark this answers is not in the document]
Comment (hamxa152) [reply .ios/Podfile]: i added back please review it
